From 97ed20be4710c2e21fb2ea9246aeacb78ba17187 Mon Sep 17 00:00:00 2001
From: Austin Shafer <ashafer@nvidia.com>
Date: Sun, 16 Oct 2022 11:42:21 -0400
Subject: [PATCH 2/8] FreeBSD: check if DRIVER_MODULE takes a devclass argument

As of 2e32d4e41d205d6f14834f87306a77ff77b9c0bd the devclass argument
was removed from DRIVER_MODULE since it was not used. This changes
checks that to avoid compilation errors on 14+.
---
 nvidia/src/nvidia/nvidia_pci.c  | 4 ++++
 nvidia/src/nvidia/nvidia_subr.c | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/nvidia/src/nvidia/nvidia_pci.c b/nvidia/src/nvidia/nvidia_pci.c
index ae2257584e3..3b5d0952003 100644
--- a/nvidia/src/nvidia/nvidia_pci.c
+++ b/nvidia/src/nvidia/nvidia_pci.c
@@ -338,7 +338,11 @@ static driver_t nvidia_pci_driver = {
     sizeof(struct nvidia_softc)
 };
 
+#if __FreeBSD_version >= 1400058
+DRIVER_MODULE(nvidia, vgapci, nvidia_pci_driver, nvidia_modevent, 0);
+#else
 DRIVER_MODULE(nvidia, vgapci, nvidia_pci_driver, nvidia_devclass, nvidia_modevent, 0);
+#endif
 MODULE_VERSION(nvidia, 1);
 
 MODULE_DEPEND(nvidia, mem, 1, 1, 1);
diff --git a/nvidia/src/nvidia/nvidia_subr.c b/nvidia/src/nvidia/nvidia_subr.c
index 43127a6afef..161e289471d 100644
--- a/nvidia/src/nvidia/nvidia_subr.c
+++ b/nvidia/src/nvidia/nvidia_subr.c
@@ -727,6 +727,9 @@ int nvidia_modevent(
              * time.
              */
             sc = &nvidia_ctl_sc;
+#if __FreeBSD_version >= 1400058
+            nvidia_devclass = devclass_create("nvidia");
+#endif
 
             nvidia_stack_t_zone = uma_zcreate("nvidia_stack_t", sizeof(nvidia_stack_t),
                     NULL, NULL, NULL, NULL, UMA_ALIGN_PTR, 0);
-- 
2.35.1

